pkg/api/v1/router_firmware.go

```diff
@@ -0,0 +1,141 @@
+package serverservice
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/volatiletech/null/v8"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+
+	"go.hollow.sh/serverservice/internal/models"
+)
+
+func (r *Router) firmwareList(c *gin.Context) {
+	pager := parsePagination(c)
+
+	var params ComponentFirmwareVersionListParams
+	if err := c.ShouldBindQuery(&params); err != nil {
+		badRequestResponse(c, "invalid filter", err)
+		return
+	}
+
+	mods := params.queryMods()
+
+	dbFirmwares, err := models.ComponentFirmwareVersions(mods...).All(c.Request.Context(), r.DB)
```

Comment:
given that there is pagination taking place... wouldn't you want to [add pagination parameters to the DB query](https://www.cockroachlabs.com/docs/stable/limit-offset.html)? Else, this is loading a potentially big number of rows in memory every time there's a query, and then discarding it once the paginated response is served. This could lead to performance degradation and increased memory usage.

Reply:
Good catch! Updated `mods` to include pagination clause.